src/app.rs

```diff
@@ -301,135 +278,67 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
-                routes: RefCell::new(routes),
-                window_event_handler_manager: RefCell::new(EventHandlerManager::new()),
-                sub_manager: RefCell::new(SubManager::new()),
-                msg_listeners: RefCell::new(Vec::new()),
-                scheduled_render_handle: RefCell::new(None),
-                after_next_render_callbacks: RefCell::new(Vec::new()),
-                render_info: Cell::new(None),
-            }),
-        }
-    }
-
     /// Bootstrap the dom with the vdom by taking over all children of the mount point and
     /// replacing them with the vdom if requested. Will otherwise ignore the original children of
     /// the mount point.
-    fn bootstrap_vdom(&self, mount_type: MountType) -> El<Ms> {
+    fn bootstrap_vdom(&self) -> El<Ms> {
         // "new" name is for consistency with `update` function.
         // this section parent is a placeholder, so we can iterate over children
         // in a way consistent with patching code.
         let mut new = El::empty(Tag::Placeholder);
 
         // Map the DOM's elements onto the virtual DOM if requested to takeover.
-        if mount_type == MountType::Takeover {
-            // Construct a vdom from the root element. Subsequently strip the workspace so that we
-            // can recreate it later - this is a kind of simple way to avoid missing nodes (but
-            // not entirely correct).
-            // TODO: 1) Please refer to [issue #277](https://github.com/seed-rs/seed/issues/277)
-            let mut dom_nodes: El<Ms> = (&self.cfg.mount_point).into();
-            #[cfg(debug_assertions)]
-            dom_nodes.warn_about_script_tags();
-
-            dom_nodes.strip_ws_nodes_from_self_and_children();
-
-            // Replace the root dom with a placeholder tag and move the children from the root element
-            // to the newly created root. Uses `Placeholder` to mimic update logic.
-            new.children = dom_nodes.children;
+        // Construct a vdom from the root element. Subsequently strip the workspace so that we
+        // can recreate it later - this is a kind of simple way to avoid missing nodes (but
+        // not entirely correct).
+        // TODO: 1) Please refer to [issue #277](https://github.com/seed-rs/seed/issues/277)
+        let mut dom_nodes: El<Ms> = (self.cfg().mount_point()).into();
+        #[cfg(debug_assertions)]
+        dom_nodes.warn_about_script_tags();
+
+        dom_nodes.strip_ws_nodes_from_self_and_children();
+
+        // Replace the root dom with a placeholder tag and move the children from the root element
+        // to the newly created root. Uses `Placeholder` to mimic update logic.
+        new.children = dom_nodes.children;
+
+        // Recreate the needed nodes.
+        // TODO: Please refer to [issue #277](https://github.com/seed-rs/seed/issues/277)
+        // @TODO_B: Check 0.7 API changes removal of MountType allows for removal of legacy solutions
```

Comment:
Depending on how `rerender_vdom` does things differently as well as the desired behavior, this is not a legacy solution. That said, if `rerender_vdom` does do what this function does, it should be fine.

To clarify regarding the desired behavior, refer to my comment about the `mount_point` circa line 180, since these two are directly related.